### PR TITLE
Update changelog for 7.0 Release

### LINKED ISSUE #642: Tracking release 7.0 changes
Since we will be delivering a major release (7.0), I'm hoping we can use this issue to track changes that might affect existing applications.  We should, at this time, go ahead and remove deprecations that have been in place as well as other api-changes (preferably minor in nature). 

- [ ] Remove deprecated [`kernel_cmd`](https://github.com/jupyter/jupyter_client/blob/96e23a843102ccb8e81aa46d1a28d8841804a8f2/jupyter_client/manager.py#L148) trait
- [ ] Remove deprecated [`cleanup()`](https://github.com/jupyter/jupyter_client/blob/96e23a843102ccb8e81aa46d1a28d8841804a8f2/jupyter_client/manager.py#L438) method
- [ ] Remove `block` parameter from [`get_msg()`](https://github.com/jupyter/jupyter_client/blob/96e23a843102ccb8e81aa46d1a28d8841804a8f2/jupyter_client/channels.py#L258)
- [ ] Make kernel manager [`request_shutdown()`](https://github.com/jupyter/jupyter_client/blob/96e23a843102ccb8e81aa46d1a28d8841804a8f2/jupyter_client/manager.py#L381) `async`
- [ ] Rename `KernelSpecManager.whitelist` trait to `allowlist`, `allowed_kernels`, ?
- [ ] Document these changes and others in the docs

We should probably talk about branching as well.  I guess we'll keep working off the tip and create a 6.x branch against the appropriate tag once a 6.x PR is required.

### PR DESCRIPTION
**Pull request by blink1073**
closes #642

### CODE REVIEW

**Comment by blink1073** [comment]
cc @davidbrochart @kevin-bates for approval, then I'll cut the release

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,46 +6,57 @@
 
 ([Full Changelog](https://github.com/jupyter/jupyter_client/compare/26a16c0c91e245f7403aa27a812fee5e905d2964...31750bc87baf88377bcc6967e227b650b38fa872))
 
+### Summary
+
+The 7.0 release brings a major feature in [Kernel Provisioners](https://github.com/jupyter/jupyter_client/blob/master/docs/provisioning.rst), which enable the ability for third parties to manage the lifecycle of a kernel's runtime environment.
+
+Being a major release, there are a some backward incompatible changes:
```
Comment (davidbrochart) [inline CHANGELOG.md:13]:
```suggestion
Being a major release, there are some backward incompatible changes:
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,46 +6,57 @@
...
+
+- Custom `KernelManager` and `KernelSpecManager` subclasses will want want to be aware of changes from the provisioners, which are detailed in the [documentation](https://github.com/jupyter/jupyter_client/blob/master/docs/provisioning.rst). (#612)
```
Comment (davidbrochart) [inline CHANGELOG.md:15]:
```suggestion
- Custom `KernelManager` and `KernelSpecManager` subclasses will want to be aware of changes from the provisioners, which are detailed in the [documentation](https://github.com/jupyter/jupyter_client/blob/master/docs/provisioning.rst). (#612)
```

**Review by davidbrochart — APPROVED** [review]

**Comment by kevin-bates** [comment]
@blink1073 - Thanks for putting this together and driving this process!  This looks good.  

I believe the specifics of the method signature changes are only documented in [this comment](https://github.com/jupyter/jupyter_client/issues/642#issuecomment-891418487) as I wasn't sure how they'd be officially coordinated. 

Does this discussion about adding a message-based interrupt into `KernelClient` warrant a delay? https://github.com/ipython/ipykernel/pull/741#issuecomment-899739225

**Comment by blink1073** [comment]
> I believe the specifics of the method signature changes are only documented in this comment as I wasn't sure how they'd be officially coordinated.

Ah, thank you, I think we can fold those changes into this change log entry.

> Does this discussion about adding a message-based interrupt into KernelClient warrant a delay? ipython/ipykernel#741 (comment)

Yes, let's hold off a bit

**Comment by blink1073** [comment]
Actually, I'll make a separate transition guide and link to it to keep the changelog more focused

**Comment by blink1073** [comment]
I've updated the changelog to refer to the new migration guide from #681.

**Comment by blink1073** [comment]
@ccordoba12, are you planning to address https://github.com/ipython/ipykernel/pull/741#issuecomment-899739225 within the next couple days?  If not we can plan to include it in a 7.1 release.

**Review by kevin-bates — APPROVED** [review]
Thank you Steve.

**Comment by blink1073** [comment]
I plan to cut 7.0 final release tomorrow

**Comment by blink1073** [comment]
Release time!

**Pull request merged**